ui/src/dashboards/reducers/ui.js

```diff
@@ -21,7 +21,21 @@ import {TEMPLATE_VARIABLE_TYPES} from 'src/tempVars/constants'
 const ui = (state = initialState, action) => {
   switch (action.type) {
     case 'LOAD_DASHBOARDS': {
-      const {dashboards} = action.payload
+      const {dashboards: dashboardsInState} = state
+      const {dashboards: loadedDashboards} = action.payload
+
+      const dashboards = loadedDashboards.map(loadedDash => {
+        const dashInState = _.find(
+          dashboardsInState,
+          d => d.id === loadedDash.id
+        )
+
+        if (dashInState) {
+          return dashInState
+        }
+        return loadedDash
```

Comment:
I think if `dashInState` we should return `{...loadedDash, templates:dashInState.templates}`